### ISSUE TITLE
A Exposed BOM (Bill of Materials) ?

### ISSUE THREAD

**Issue by scorsi** [JetBrains/Exposed#966]
Hello,

It could be cool to have a Maven BOM to automatically manage Exposed versions.
Look here: https://docs.spring.io/dependency-management-plugin/docs/current-SNAPSHOT/reference/html/

Thanks ! :)

**Comment by ps-feng** [comment]
Since the project already uses Gradle, we could use a [platform](https://docs.gradle.org/current/userguide/platforms.html) instead and avoid a dependency on that plugin.

**Comment by scorsi** [comment]
Maybe other people don’t use Gradle ?

BOM are supported by all build system included Gradle without the need of the Spring dependency management plugin. 

And if by dependency you’re talking about one single Pom of around some KB which isn’t included in the final build even with ShadowJar, so yes, a BOM include a dependency.

**Comment by ps-feng** [comment]
Agreed, I meant to use Gradle's built-in [java-platform plugin](https://docs.gradle.org/current/userguide/java_platform_plugin.html#sec:java_platform_publishing), which can publish BOMs.

As for the Gradle platform vs Spring plugin, I'd be interested in hearing what problems you ran into as we replaced it completely without any issue (I must admit it was more difficult than it should because the documentation is not clear enough and I couldn't find any sample).

**Comment by scorsi** [comment]
I'm the only developer on a very large project, I haven't the workload to handle the transition to be honest. I pretty like the `dependencyManagement` task separated from the `dependencies` one. I felt into one issue with the project which is a mono-repo with about 30 subprojects (serverless micro-services or shared-part of micro-services for the majority) I use the Spring plugin like this:
```gradle
dependencyManagement {
    imports {
        mavenBom "com.amazonaws:aws-java-sdk-bom:1.11.805"
        mavenBom "com.fasterxml.jackson:jackson-bom:2.11.0"
        // ... truncated, other boms
    }
    dependencies {
        dependency "org.jetbrains.kotlinx:kotlinx-coroutines-core:1.3.7"
        dependency "com.amazonaws:aws-lambda-java-core:1.2.1"
        // ... truncated, lot of other deps
    }
}
```
Example of a subproject build.gradle:
```gradle
dependencies {
    implementation "org.postgresql:postgresql"
    implementation "com.zaxxer:HikariCP"
    // others deps ...
}
```
Which helps me manage all deps versions for every 30 projects. All subprojects will recover the versions from the main one.
I also use dependabots and it'll just bump version in that part not for every subprojects. It simpler to manage as the only-one devs.
I didn't found how I could achieve that, maybe it can be achieved.

**Comment by ps-feng** [comment]
It's pretty much the same with Gradle platforms. It works something like this:
1. Create a module (folder), say `bom`.
2. Create a `build.gradle.kts` file inside (equivalent to `dependencyManagement`):
```kotlin
plugins {
    `java-platform`
}

javaPlatform {
    allowDependencies()
}

dependencies {
    api(platform("com.amazonaws:aws-java-sdk-bom:1.11.805"))
    api(platform("com.fasterxml.jackson:jackson-bom:2.11.0"))

    constraints {
        api("org.jetbrains.kotlinx:kotlinx-coroutines-core:1.3.7")
        api("com.amazonaws:aws-lambda-java-core:1.2.1")
    }
}
```
3. Add it to `settings.gradle.kts`
```kotlin
include(":bom")
```
4. Use in any subproject:
```kotlin
implementation(platform(project(":bom")))  // you could inject this from the root build.gradle.kts if you don't want to put it in all projects

implementation("org.postgresql:postgresql")
implementation("com.zaxxer:HikariCP")
```
So as you can see, it's pretty much equivalent.

**Comment by scorsi** [comment]
Thanks for the solution !
It will of course require time but I don't need to investigate any-more now thanks for the tips ! It will be much more clearer for sure ! I'm managing a `build.gradle` file for around 200 LOC (around 40 LOC of deps), it's easy to be lost in it ^^
I will add this on the to-do tasks (with a very low priority since it's not fundamental).

**Comment by Tapac** [comment]
Fixed in master. Will be available in 0.32.1

**Comment by Tapac** [comment]
Released in 0.32.1. Check https://github.com/JetBrains/Exposed/tree/master/exposed-bom
